modyn/selector/internal/selector_strategies/abstract_selection_strategy.py

```diff
@@ -398,3 +398,20 @@ def _persist_samples(self, keys: list[int], timestamps: list[int], labels: list[
 
         for proc in processes:
             proc.join()
+
+    def get_available_labels(self) -> list[int]:
+        with MetadataDatabaseConnection(self._modyn_config) as database:
+            result = (
+                database.session.query(SelectorStateMetadata.label)
+                .filter(
+                    SelectorStateMetadata.pipeline_id == self._pipeline_id,
+                    SelectorStateMetadata.seen_in_trigger_id >= self._next_trigger_id - self.tail_triggers
+                    if self.tail_triggers is not None
+                    else True,
+                )
+                .distinct()
+                .all()
+            )
+            available_labels = [result_tuple[0] for result_tuple in result]
```

Comment:
I suppose it is not a problem that these lists all labels that have been seen and it could be (e.g. due to random presampling) that not all labels are actually part of the TriggerTrainingSet in the end?

Reply:
Not a problem, simply no points are returned

Reply:
To actually know which classes are selected, we could have to use more inefficient presampling methods (like `last used in trigger` as in the other PR). I don't think it is worth it.